File: tasks5/CharacterTree.c
//
// Created by Sebastian Nielebock on 03.04.2019.
//

#include <stdio.h>

/*
 *       *
 *      ***
 *     *****
 *    *******
 *       *
 */

void print_tree_char(int height, char sym){
    int startSpaces = height-1;
    int numSpaces = startSpaces;
    int numSym = 1;
    int i;
    for(i = 0; i<height;i++){
        int sp, sy;
        for(sp = 0; sp < numSpaces;sp++){
            printf(" ");
        }
        for(sy = 0; sy < numSym; sy++){
            printf("%c",sym);
        }
        numSpaces--;
        numSym+=2;
        printf("\n");
    }
    int lsp;
    for(lsp = 0; lsp < startSpaces;lsp++){
        printf(" ");
    }
    printf("*\n\n");
}

void print_tree(int height){
    print_tree_char(height, '*');
}

int main(void){
    print_tree(5);
    print_tree_char(5,'o');
    return 0;
    print_tree(1);
    print_tree(2);
    print_tree(3);
    print_tree(4);
    print_tree(10);
    print_tree_char(10,'+');
    print_tree_char(10,'o');
    return 0;
}

File: pgp_scala/src/main/scala/CharacterTree.scala
object CharacterTree {

      def rec_tree(numSpaces: Int,numSym:Int, height: Int, char:Char): Unit ={
        if(height == 0){
          return
        }
        print_level(numSpaces,' ')
        print_level(numSym,char)
        println()
        rec_tree(numSpaces-1,numSym+2,height-1,char)
      }

      def print_level(level:Int,char:Char): Unit ={
        if(level == 0){
          return
        }
        print(char)
        print_level(level-1,char)
      }

      def print_tree(height: Int) {
        print_tree_char(height,'*')
      }
      
      def print_tree_char(height: Int, char: Char) {
        rec_tree(height-1,1,height,char)
        rec_tree(height-1,1,1,char)
      }

      def main(args: Array[String]) {
        var foo: Int = 14
        print_tree(foo)
        print_tree_char(foo, 'o');
      }
   }
